waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -150,6 +155,8 @@ proc connectedness*(pm: PeerManager, peerId: PeerId): Connectedness =
   if (storedInfo == StoredInfo()):
     # Peer is not managed, therefore not connected
     return NotConnected
+  elif peerId in pm.blacklist:
+    return ShouldNotConnect
```

Comment:
`NotConnected` is a special case since it means the peer is not in the peer store. 
`ShouldNotConnect` should probably be updated in the `connectionBook` and read from there.